src/components/TransactionHistoryRows/TransactionContract.tsx

```diff
@@ -35,18 +39,31 @@ export const TransactionContract = ({
     isReceive && !txDetails.tx.data?.method ? txDetails.tradeTx?.type : txDetails.tx.data?.method
   const isFirstAssetOutgoing = interactsWithWithdrawMethod && isSend
 
+  const isRevoke = useMemo(
+    () =>
+      Boolean(
+        txMetadata?.method === 'approve' && txMetadata.value && bn(txMetadata.value).isZero(),
```

Comment:
Note: Not using`bnOrZero` is on purpose, missing value and `'0'` value  are two different things, one means a revoke, the other literally means we didn't parse any `value`